libknet/link.c

```diff
@@ -65,6 +67,12 @@ int knet_link_set_config(knet_handle_t knet_h, uint16_t host_id, uint8_t link_id
 		return -1;
 	}
 
+	if (transport != KNET_TRANSPORT_UDP && transport != KNET_TRANSPORT_SCTP &&
+	    transport != KNET_TRANSPORT_TCP) {
+		errno = EINVAL;
```

Comment:
Not urgent, but let´s change this to be more flexible. if transport >= KNET_MAX_TRANSPORTS will survive a long time :)